app/controllers/benefits/reimbursed/NonTaxableCostsBenefitsAmountController.scala

```diff
@@ -0,0 +1,117 @@
+  )
+
+  private def redirects(cya: EmploymentCYAModel, taxYear: Int, employmentId: String) = {
+    RedirectService.expensesAmountRedirects(cya, taxYear, employmentId)
+  }
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+      employmentSessionService.getAndHandle(taxYear, employmentId) { (optCya, prior) =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, optCya, EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+          val cyaAmount = cya.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expenses))
+
+          val form = fillFormFromPriorAndCYA(amountForm, prior, cyaAmount, employmentId)(
+            employment =>
+              employment.employmentBenefits.flatMap(_.benefits.flatMap(_.expenses))
+          )
+          Future.successful(Ok(nonTaxableCostsAmountBenefitsView(taxYear, form, cyaAmount, employmentId)))
+        }
+      }
+    }
+
+  }
+
+  def submit(taxYear: Int, employmentId: String): Action[AnyContent] = authAction.async { implicit user =>
+    inYearAction.notInYear(taxYear) {
+
+      val redirectUrl = CheckYourBenefitsController.show(taxYear, employmentId).url
+
+      employmentSessionService.getSessionDataAndReturnResult(taxYear, employmentId)(redirectUrl) { cya =>
+
+        redirectBasedOnCurrentAnswers(taxYear, employmentId, Some(cya), EmploymentBenefitsType)(redirects(_, taxYear, employmentId)) { cya =>
+
+          amountForm.bindFromRequest().fold(
+            { formWithErrors =>
+              val cyaAmount = cya.employment.employmentBenefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel.flatMap(_.expenses))
+              Future.successful(BadRequest(nonTaxableCostsAmountBenefitsView(taxYear, formWithErrors, cyaAmount, employmentId)))
+            }, {
+              amount =>
+                val cyaModel = cya.employment
+                val benefits = cyaModel.employmentBenefits
+                val reimbursedCostsVouchersAndNonCashModel = benefits.flatMap(_.reimbursedCostsVouchersAndNonCashModel)
+
+                val updatedCyaModel = cyaModel.copy(
+                  employmentBenefits = benefits.map(_.copy(reimbursedCostsVouchersAndNonCashModel =
+                    reimbursedCostsVouchersAndNonCashModel.map(_.copy(expenses = Some(amount)))))
+                )
+
+                employmentSessionService.createOrUpdateSessionData(employmentId, updatedCyaModel, taxYear,
+                  isPriorSubmission = cya.isPriorSubmission, hasPriorBenefits = cya.hasPriorBenefits)(errorHandler.internalServerError()) {
+
+                  if (cya.isPriorSubmission) {
+                    Redirect(CheckYourBenefitsController.show(taxYear, employmentId))
+                  } else {
+                    //TODO - redirect to taxable expenses yes no question
```

Comment:
this page has been created, TaxableCostsBenefitsController, if you want to link it there?

Reply:
done